[REM-25138][iOS] [Perf] Debug builds should always send performance data if config API response is valid

shivaiahmahesha opened the pull request:
Changed the info.plist key name,
Wrapped it around the debug block.

donnie-jp committed:
Feature/rem 23675 threadblock (#2)

REM-23675 (feat) Add watcher for main thread being blocked longer than a threshold value

* fix unit tests by setting a valid configuration in test setup
* fix exception bug where we could pass a nil string argument to `containsString` method

vito1188 committed:
Enable battery monitoring

shivaiahmahesha committed:
[REM-24598][iOS] [Perf] Use AppID instead of package name

donnie-jp committed:
REM-23675 Disable watcher thread when tracking gets stopped

donnie-jp committed:
Add guard around use of UIControlEventPrimaryActionTriggered because it is only available from iOS 9.0 (#8)

shivaiahmahesha committed:
REM-25013 Include AppID in payload sent to event hub

donnie-jp committed:
REM-25128 Prepare for publishing to CocoaPods (#10)

- use a "-github" suffix on the version tag to differentiate with the internal repo's tags
- update pod installation instructions

shivaiahmahesha committed:
[REM-25138][iOS] [Perf] Debug builds should always send performance data if config API response is valid

donnie-jp committed:
REM-25817 Fix crash that occurs when info.plist keys are missing by failing safely in release builds (#13)

donnie-jp commented on Tests/ConfigTests.m:
```diff
@@ -167,7 +167,7 @@ - (void)testThatTrackingStoppedWhenStubbedInvalidConfig
 }
 
-- (void)testThatTrackingStoppedWhenUpdatedConfigActivationRatioIsLowerThanPreviousValue
+- (void)testIrrespectiveOfConfigActivationRatioDebugBuildsAlwaysSendData
```
testIrrespectiveOfConfigActivationRatioDebugBuildsAlwaysSendData -> testThatIrrespectiveOfConfigActivationRatioSenderIsRunningInDebugBuilds

donnie-jp replied:
Please also add a test `testThatSenderIsRunningWhenConfigActivationRatioIsZero`

shivaiahmahesha committed:
[REM-25138]Adding more test case

donnie-jp committed:
feat(swizzling): REM-24817 Rewrite WKWebView swizzling (#14)

- use IMP blocks for swizzling (instead of the approach where new selector methods were created on categories) so that selectors are unchanged
- move the force tracking check to before init’s disable tracking call
- remove unit tests that relied on the old selector IMP swapping approach
- remove WKWebView swizzling category because it is incompatible with the new approach
- fix pragma typos in tests source
- remove doxygen `Standard metrics` group because we don't have exposed standard metrics now

shivaiahmahesha committed:
Merge branch 'master' into REM-25138
